pkg/segment/query/processor/dataprocessor.go

```diff
@@ -250,6 +258,7 @@ func NewBinDP(options *structs.BinCmdOptions) *DataProcessor {
 		isPermutingCmd:    false,
 		isBottleneckCmd:   !hasSpan,
 		isTwoPassCmd:      !hasSpan,
+		processorLock:     &sync.Mutex{},
```

Comment:
nit: you might be able to change the type from `*Mutex` to `Mutex` and then you don't need to add this line to every constructor. But feel free to keep what you have

Reply:
if I change the mutex to `Mutex`, then there is an issue with `copylocks`. This is the error `copylocks: literal copies lock value from *scrollerDP`

So apparently it creates copy of the lock when we pass or assign it another reference making the locks not work properly.